backend/request.go

```diff
@@ -86,6 +87,55 @@ func GetHost(r *http.Request) string {
 	return "https://" + r.Host
 }
 
+func stringMap(vs []string, f func(string) string) []string {
+	vsm := make([]string, len(vs))
+	for i, v := range vs {
+		vsm[i] = f(v)
+	}
+	return vsm
+}
+
+// Converts header entries associated with key to canonical form. In particular,
+// multiple headers are collapsed into one.
+func canonical(h http.Header, key string) {
+	v := h[textproto.CanonicalMIMEHeaderKey(key)]
+	var a []string
+	for _, vv := range v {
+		if vv != "" {
+			a = append(a, stringMap(strings.Split(vv, ","), strings.TrimSpace)...)
```

Comment:
just a nit (feel free to ignore): I prefer methods to be ordered in call order (e.g. stringMap goes below canonical). That makes it easier to understand what's going on.

Reply:
I have no preference; changed in e553873.